### ISSUE TITLE
IPFS tabs (still) sometimes fail to reload in Session Restore

### ISSUE THREAD

**Issue by stephendonner** [brave/brave-browser#14328]
## Description 
Note: This is a follow-up bug to the landed https://github.com/brave/brave-browser/issues/13932 (which is itself a followup to https://github.com/brave/brave-browser/issues/13519), so the steps are pretty much the same, but filing so the fix(es) can be cleaner.  (Was asked to file this issue anew, rather than reopen 13932.)

With multiple `ipfs://` or `ipns://` URLs open in the background (with 1 active), upon browser restart (Session Restore), sometimes the interstitial pages will show, instead of the tabs loading their content.

## Steps to Reproduce

pre-req: have IPFS installed

1. open the following 3 ipfs/ipns pages such as: `ipfs://bafybeiemxf5abjwjbikoz4mc3a3dla6ual3jsgpdr4cjr3oz3evfyavhwq/wiki/Vincent_van_Gogh.html`, `ipfs://bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi/`, `ipns://k2k4r8oj77eqrr4018u3qjk2byycrz4pmxyodaruuqv4zvph6foed004/`
2. Select the middle tab
3. Relaunch
4. While the active IPFS tab is loading, click the other 2.

## Actual result:

![ipfs-tabs](https://user-images.githubusercontent.com/387249/108905853-0fe3c000-75d5-11eb-90cc-29327a8a2134.gif)

## Expected result:

IPFS should poll/retry connecting to peers, and reloading the tabs, in the background, until successful (or we reach another error state/timeout?)

## Reproduces how often: 
85%? - pretty often, at least

## Brave version (brave://version info)

Brave | 1.22.42 Chromium: 89.0.4389.58 (Official Build) nightly (x86_64)
-- | --
Revision | 1a139f28ecc27719439e37c6b1533cee999cb802-refs/branch-heads/4389@{#1134}
OS | macOS Version 11.2.1 (Build 20D74)

## Version/Channel Information:

- Can you reproduce this issue with the current release? yes
- Can you reproduce this issue with the beta channel? yes
- Can you reproduce this issue with the dev channel?  yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
- Does the issue resolve itself when disabling Brave Rewards? 
- Is the issue reproducible on the latest version of Chrome? 

## Miscellaneous Information:

cc: @spylogsster @LaurenWags

**Comment by GeetaSarvadnya** [comment]
Reproduced the issue in Windows 10 x64

```

Brave | 1.22.50 Chromium: 89.0.4389.72 (Official Build) dev (64-bit)
-- | --
Revision | 3f345f156bfd157bd1bea06310e55f3fb2490359-refs/branch-heads/4389@{#1393}
OS | Windows 10 OS Version 2004 (Build 19041.804)


```
@stephendonner Does ipns link working for you?

**Comment by stephendonner** [comment]
@GeetaSarvadnya nope; I have replaced it now with `ipns://brantly.eth` which currently works 👍

**Comment by GeetaSarvadnya** [comment]
Verification passed on

```
Brave | 1.23.41 Chromium: 89.0.4389.90 (Official Build) nightly (64-bit)
-- | --
Revision | 62eb262cdaae9ef819aadd778193781455ec7a49-refs/branch-heads/4389@{#1534}
OS | Windows 10 OS Version 2004 (Build 19041.867)

```
- Verified the STR from the description and ensured that after session restores interstitial page isn't shown upon clicking on each tab and confirmed that tab content is shown correctly

**Comment by stephendonner** [comment]
Note: when testing this -- at least on `macOS` -- it might help to do so if/when https://github.com/brave/brave-browser/issues/14816 is fixed, first 👍
